### PR TITLE
/history/by_task_id : a method to link each task's history uniquely to the SQL database

### PR DESCRIPTION
**Pull request by tashrifbillah**
I am a visionary that has been utilizing Luigi for medical research since 2020. Our users are non-programmers who need convenience over complexity to examine a task's history. I scrutinized your [db_task_history.py](https://github.com/spotify/luigi/blob/master/luigi/db_task_history.py) and [server.py](https://github.com/spotify/luigi/blob/master/luigi/server.py) and figured out how it could made easy for non-programmers. My proposed method has been working since 2020. To complete this PR, I have enriched the documentation and added screenshot.

## Description

No change in the existing functionality. Only added `/history/by_task_id/` endpoint.

## Motivation and Context

The existing method for getting a single task's history in future is not adequate. To get so, one has to remember the [numeric id](https://github.com/spotify/luigi/blob/master/doc/central_scheduler.rst#enabling-task-history) of a task. This PR omits that burden by providing a method to autonomously obtain a single task's history. As a result, we can now construct history trees and link each task's history uniquely to the SQL database.

## Have you tested this? If so, how?

I ran my jobs with this code and it works for me.

### CODE REVIEW

**Comment by tashrifbillah** [comment]
Hi @PeteW , I hope you are well. We interacted last in 2020. This is the remnant of what came out of that interaction. Being inspired by the [tree functionality](https://github.com/spotify/luigi/blob/829fc0c36ecb4d0ae4f0680dec6d538577b249a2/luigi/tools/deps_tree.py), I provided a way to autonomously obtain a single task's history in order to construct an HTML tree. Can you please review it?

**Comment by PeteW** [comment]
Good to hear from you! Yes I remember. I'll have a response for you by tomorrow

**Comment by PeteW** [comment]
This is great. The coverage tests indicate a warning about adding small methods without coverage. However from my standpoint this is a nitpick. The change makes sense and thanks for bringing me back to review this great project.

**Review by PeteW — APPROVED** [review]

File: luigi/db_task_history.py [hunk 1]
```diff
@@ -187,6 +187,13 @@ def find_task_by_id(self, id, session=None):
         with self._session(session) as session:
             return session.query(TaskRecord).get(id)
 
+    def find_task_by_task_id(self, task_id, session=None):
+        """
+        Find task with the given task ID.
+        """
+        with self._session(session) as session:
+            return session.query(TaskRecord).filter(TaskRecord.task_id==task_id).all()[-1]
```
Comment (dlstadther) [inline luigi/db_task_history.py:195]: flake8 tests complain about this line.

```
./luigi/db_task_history.py:195:71: E225 missing whitespace around operator
```

**Review by dlstadther — COMMENTED** [review]
Please look at the failing flake8 test.

Also @RRap0so , would you or another admin of the repo be able to look into why PRs keep expecting results from the `base (3.9, docs)` and `base (3.9, flake8)` tests (these were replaced by newer tests). Based on a quick search, it sounds like there might be branch protection rules specified on `master` which requires series of specific test successes. I don't have access to the luigi `Settings` page to check for myself.

File: luigi/db_task_history.py (hunk 1 above)
Comment (tashrifbillah) [reply luigi/db_task_history.py:195]: @dlstadther , could you tell us what needs to be corrected here? Should we put space around `==`?

Comment (RRap0so) [reply luigi/db_task_history.py:195]: @dlstadther done. Anything that needs to be added?

@tashrifbillah that should be it.

Comment (dlstadther) [reply luigi/db_task_history.py]: Thanks @RRap0so !

**Review by dlstadther — APPROVED** [review]

**Comment by tashrifbillah** [comment]
Thank you all. Just installed the master branch and verified that all functionalities are intact!

**Pull request merged**